x/mock/simulation/random_simulate_blocks.go

```diff
@@ -239,15 +242,22 @@ func getBlockSize(r *rand.Rand, blockSize int) int {
 }
 
 // adds all future operations into the operation queue.
-func queueOperations(queuedOperations map[int][]Operation, futureOperations []FutureOperation) {
+func queueOperations(queuedOperations map[int][]Operation, queuedTimeOperations []FutureOperation, futureOperations []FutureOperation) {
 	if futureOperations == nil {
 		return
 	}
 	for _, futureOp := range futureOperations {
-		if val, ok := queuedOperations[futureOp.BlockHeight]; ok {
-			queuedOperations[futureOp.BlockHeight] = append(val, futureOp.Op)
+		if futureOp.BlockHeight != 0 {
+			if val, ok := queuedOperations[futureOp.BlockHeight]; ok {
+				queuedOperations[futureOp.BlockHeight] = append(val, futureOp.Op)
+			} else {
+				queuedOperations[futureOp.BlockHeight] = []Operation{futureOp.Op}
+			}
 		} else {
-			queuedOperations[futureOp.BlockHeight] = []Operation{futureOp.Op}
+			index := sort.Search(len(queuedTimeOperations), func(i int) bool { return queuedTimeOperations[i].BlockTime.After(futureOp.BlockTime) })
```

Comment:
Since we have to do a search, can you make a TODO comment to make the queued time ops a sorted data structure? (This can then be addressed #postlaunch or if it bottlenecks the simulation)

Right now we have to copy all the time ops on insertion / deletion, which is an O(N) operation.